src/org/deepsymmetry/bytefield/core.cljs

```diff
@@ -512,6 +512,52 @@
   ([address label attr-spec]
    (draw-related-boxes (repeat (- address (next-address)) label) attr-spec)))
 
+(defn- draw-gap-trapezoid
+  "A private function used by draw-gap and draw-inline-gap for drawing
+  trapezoids. The first four arguments specifies the trapezoid's
+  vertices. The fifth argument is the trapezoid's 'type' e.g. upper, 
+  lower, left or right. This type is used to calculate on which edge
+  the 'gap' should appear. The sixth argument is the fill parameter 
+  for the trapezoid. The seventh parameter is the specification for
+  styling the gap."  
+  ([top-left top-right bottom-right bottom-left trapezoid-type fill gap-style]
+   (when fill 
+     (append-svg (svg/polygon (concat
+       top-left
+       top-right
+       bottom-right
+       bottom-left
+       ) :fill fill))
+    )
+  (if (or (= trapezoid-type :trapezoid-type/upper) (= trapezoid-type :trapezoid-type/lower))
+    [
+      (apply draw-line (concat top-left bottom-left))
+      (apply draw-line (concat top-right bottom-right))
+      (if(= trapezoid-type :trapezoid-type/upper)
+        (apply draw-line (concat bottom-left bottom-right [gap-style]))
+        (if(= trapezoid-type :trapezoid-type/lower)
```

Comment:
This refactoring is a great idea which makes an excellent contribution even better. I’m going to merge it now since I’ve already used so much of your time, and the only tweaks I am going to make are changing the `if` expressions which have empty function invocations as their “else” branches to `when` expressions, which is the more idiomatic way of expressing an “if” with no “else”, and to apply Clojure-style grouping of the parentheses, brackets, and braces.

Reply:
In case you’re interested, in my final tweaks I also changed a couple of places where there was an `or` to check the trapezoid type to use the slightly more idiomatic approach of using a set as a function to check membership. This is terse and a bit opaque to new Clojure developers but becomes concise and clear with experience. For example:
```clojure
(if (#{:trapezoid-type/upper :trapezoid-type/lower} trapezoid-type)
  …)
```